core/comp_ui.py

```diff
@@ -31,17 +30,22 @@
 
 
 def _PromptLen(prompt_str):
-  """Ignore all characters between \x01 and \x02."""
+  """Ignore all characters between \x01 and \x02 and handle unicode characters.
+  In particular, the display width of a string may be different from either the
+  number of bytes or the number of unicode characters."""
   escaped = False
   length = 0
+  chunk = ""
   for c in prompt_str:
     if c == '\x01':
       escaped = True
+      length += libc.wcswidth(chunk)
```

Comment:
RuntimeError isn't handled.  Maybe we should fall back on the old calculation in that case?